modules/gapi/src/backends/common/gbackend.hpp

```diff
@@ -46,9 +46,11 @@ namespace magazine {
 
 } // namespace magazine
 #if !defined(GAPI_STANDALONE)
-using Mag = magazine::Class<cv::gapi::own::Mat, cv::UMat, cv::gapi::own::Scalar, cv::detail::VectorRef, cv::detail::OpaqueRef>;
+using Mag = magazine::Class<cv::gapi::own::Mat, cv::UMat, cv::gapi::own::Scalar, cv::detail::VectorRef, cv::detail::OpaqueRef,
+                            cv::gapi::own::RMat::HostMatP>;
 #else
-using Mag = magazine::Class<cv::gapi::own::Mat, cv::gapi::own::Scalar, cv::detail::VectorRef, cv::detail::OpaqueRef>;
+using Mag = magazine::Class<cv::gapi::own::Mat, cv::gapi::own::Scalar, cv::detail::VectorRef, cv::detail::OpaqueRef,
+                            cv::gapi::own::RMat::HostMatP>;
```

Comment:
So why it stores `HostMat` instead of `RMat`??

Reply:
I didn't touch executor's interaction with backends and islands, they still know nothing about RMat and work as they work previously. Executor accesses RMat, extracts own::Mat from it and pass extracted own::Mat to islands

Reply:
They should start supporting it natively, this is the key.